README.md

```diff
@@ -5,754 +5,864 @@
+
+Let's consider the implication of this for a minute. JavaScript objects are mutable. If you create an instance of a JavaScript class, the created objects are mutable. 
+
+**🤔 How does an immutable class instance work?**
+
+This is the not the question that we set out to answer when we started working on Microstates in 2016, but it's one of the question that we believe Microstates is answering.
 
-## microstate constructor
+The challenge with answering this question is that it uproots a lot of what we know about working with JavaScript classes. Let's consider some of these and see where it takes us.
 
-Microstate constructor creates a microstate. It accepts two arguments, `Type` class that describes the structure of your state and `value` that is the initial state.
+### Instance Methods
+
+Instance methods are functions that can modify properties of the object. For example, `setFirstName` might modify the `firstName` on an instance of `Person`.
 
 ```js
-// initial value is undefined so state will be default value of String which is an empty string
-create(String).state;
-// => ''
+/* regular class */
+class Person {
+  constructor({ firstName, lastName }) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
 
-// initial value is 'hello world' so state will be hello world.
-create(String, 'hello world').state;
-// => 'hello world'
+  setFirstName(firstName) {
+    this.firstName = firstName;
+  }
+}
+
+let homer = new Person({ firstName: 'Homer', lastName: 'Simpson' });
+
+homer.setFirstName('Homer J');
 ```
 
-The object returned from the constructor has all of the transitions for the structure.
+But what if the instance is immutable, `setFirstName` must return a new instance of `Person` class without modifying the original object. This raises another question, how do you create a copy of an instance of a class? I won't answer this question here, but I'll just say that Microstates does it.
+
+Let's see what this would look like with a Microstate.
 
 ```js
-import { create } from 'microstates';
+/* microstate class type */ // these comments are FYI, they have no code purpose
+class Person {
+  firstName = String;
+  lastName = String;
 
-let ms = create(Object);
+  setFirstName(firstName) {
+    return this.firstName.set(firstName);
+  }
+}
+
+import { create } from 'microstates';
 
-console.log(ms);
-//  {
-//    assign: Function
-//    set: Function
-//  }
+let homer = create(Person, { firstName: 'Homer', lastName: 'Simpson' });
 
-ms.assign({ hello: 'world', hi: 'there' }).state;
-// => { hello: 'world', hi: 'there' }
+// original homer is unmodified
+let homerJ = homer.setFirstName('Homer J');
 ```
 
-## Composition
+Unlike a regular JavaScript class methods which imperatively mutate objects they have access to, Microstates types declaratively describe what should be different in the copy that's returned from the method.
```

Comment:
Unlike regular JavaScript class methods, which imperatively mutate objects they have access to,